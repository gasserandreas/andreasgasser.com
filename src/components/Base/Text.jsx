import React from 'react';
import styled from 'styled-components';

import { Text } from 'rebass/styled-components';

const StyledText = styled(Text)(({ theme }) => `
  color: ${theme.colors.text};
`);

export default (props) => (
  <StyledText color="text" {...props} />
);
