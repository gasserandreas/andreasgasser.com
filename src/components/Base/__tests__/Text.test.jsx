/* global withTheme */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Text from '../Text';

describe('components/Text', () => {
  it('small Text', () => {
    const dom = mount(withTheme(<Text variant="small">Small Text</Text>));
    expect(toJson(dom)).toMatchSnapshot();
  });

  it('bold Text', () => {
    const dom = mount(withTheme(<Text variant="bold">Bold Text</Text>));
    expect(toJson(dom)).toMatchSnapshot();
  });

  it('default Text', () => {
    const dom = mount(withTheme(<Text variant="default">Default Text</Text>));
    expect(toJson(dom)).toMatchSnapshot();
  });

  it('muted Text', () => {
    const dom = mount(withTheme(<Text variant="muted">Muted Text</Text>));
    expect(toJson(dom)).toMatchSnapshot();
  });
});
