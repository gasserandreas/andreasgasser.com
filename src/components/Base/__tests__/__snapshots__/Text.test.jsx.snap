// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Text bold Text 1`] = `
<CustomThemeProvider
  dark={false}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "320px",
          "540px",
          "720px",
          "960px",
          "1140px",
          "1400px",
        ],
        "buttons": Object {
          "outline": Object {
            "bg": "transparent",
            "boxShadow": "inset 0 0 2px",
            "color": "primary",
            "variant": "buttons.primary",
          },
          "primary": Object {
            "bg": "primary",
            "borderRadius": "default",
            "color": "background",
            "fontSize": 2,
            "fontWeight": "bold",
          },
          "secondary": Object {
            "bg": "secondary",
            "color": "background",
            "variant": "buttons.primary",
          },
        },
        "colors": Object {
          "background": "#fff",
          "gray": "#dddddf",
          "highlight": "hsla(205, 100%, 40%, 0.125)",
          "muted": "#f6f6f9",
          "primary": "#07c",
          "secondary": "#30c",
          "text": "#000",
        },
        "fontSizes": Array [
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
        ],
        "fontWeights": Object {
          "body": 400,
          "bold": 900,
          "heading": 700,
        },
        "fonts": Object {
          "body": "Open Sans, Helvetica Neue, Helvetica, sans-serif",
          "heading": "inherit",
          "monospace": "Menlo, monospace",
        },
        "lineHeights": Object {
          "body": 1.5,
          "heading": 1.125,
        },
        "radii": Object {
          "circle": 99999,
          "default": 4,
        },
        "shadows": Object {
          "card": "0 0 4px rgba(0, 0, 0, .125)",
        },
        "sizes": Object {
          "avatar": 48,
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "styles": Object {
          "root": Object {
            "fontFamily": "body",
            "fontWeight": "body",
            "lineHeight": "body",
          },
        },
        "text": Object {
          "bold": Object {
            "color": "#595959",
            "fontWeight": "900",
          },
          "default": Object {
            "color": "#595959",
          },
          "muted": Object {
            "color": "#808080",
          },
          "page": Object {
            "fontSize": 4,
            "letterSpacing": "-0.04em",
            "lineHeight": "1.4em",
            "wordSpacing": "-0.08em",
          },
          "section": Object {
            "fontSize": 3,
            "lineHeight": "1.4em",
          },
          "small": Object {
            "color": "#595959",
            "fontSize": 0,
          },
          "subPage": Object {
            "color": "#808080",
            "fontSize": 3,
            "lineHeight": "1.6em",
          },
        },
        "variants": Object {
          "avatar": Object {
            "borderRadius": "circle",
            "height": "avatar",
            "width": "avatar",
          },
          "card": Object {
            "bg": "background",
            "boxShadow": "card",
            "p": 2,
          },
          "link": Object {
            "color": "primary",
          },
          "nav": Object {
            ":hover,:focus,.active": Object {
              "color": "primary",
            },
            "color": "inherit",
            "display": "inline-block",
            "fontSize": 1,
            "fontWeight": "bold",
            "p": 2,
            "textDecoration": "none",
          },
        },
      }
    }
  >
    <Text
      variant="bold"
    >
      <ForwardRef
        variant="bold"
      >
        <styled.div
          tx="text"
          variant="bold"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "kfuSjk",
                  "rules": Array [
                    "box-sizing: border-box;",
                    "margin: 0;",
                    "min-width: 0;",
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "usesTheme": true,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            tx="text"
            variant="bold"
          >
            <div
              className="sc-bdVaJa kfuSjk"
            >
              Bold Text
            </div>
          </StyledComponent>
        </styled.div>
      </ForwardRef>
    </Text>
  </ThemeProvider>
</CustomThemeProvider>
`;

exports[`components/Text default Text 1`] = `
<CustomThemeProvider
  dark={false}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "320px",
          "540px",
          "720px",
          "960px",
          "1140px",
          "1400px",
        ],
        "buttons": Object {
          "outline": Object {
            "bg": "transparent",
            "boxShadow": "inset 0 0 2px",
            "color": "primary",
            "variant": "buttons.primary",
          },
          "primary": Object {
            "bg": "primary",
            "borderRadius": "default",
            "color": "background",
            "fontSize": 2,
            "fontWeight": "bold",
          },
          "secondary": Object {
            "bg": "secondary",
            "color": "background",
            "variant": "buttons.primary",
          },
        },
        "colors": Object {
          "background": "#fff",
          "gray": "#dddddf",
          "highlight": "hsla(205, 100%, 40%, 0.125)",
          "muted": "#f6f6f9",
          "primary": "#07c",
          "secondary": "#30c",
          "text": "#000",
        },
        "fontSizes": Array [
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
        ],
        "fontWeights": Object {
          "body": 400,
          "bold": 900,
          "heading": 700,
        },
        "fonts": Object {
          "body": "Open Sans, Helvetica Neue, Helvetica, sans-serif",
          "heading": "inherit",
          "monospace": "Menlo, monospace",
        },
        "lineHeights": Object {
          "body": 1.5,
          "heading": 1.125,
        },
        "radii": Object {
          "circle": 99999,
          "default": 4,
        },
        "shadows": Object {
          "card": "0 0 4px rgba(0, 0, 0, .125)",
        },
        "sizes": Object {
          "avatar": 48,
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "styles": Object {
          "root": Object {
            "fontFamily": "body",
            "fontWeight": "body",
            "lineHeight": "body",
          },
        },
        "text": Object {
          "bold": Object {
            "color": "#595959",
            "fontWeight": "900",
          },
          "default": Object {
            "color": "#595959",
          },
          "muted": Object {
            "color": "#808080",
          },
          "page": Object {
            "fontSize": 4,
            "letterSpacing": "-0.04em",
            "lineHeight": "1.4em",
            "wordSpacing": "-0.08em",
          },
          "section": Object {
            "fontSize": 3,
            "lineHeight": "1.4em",
          },
          "small": Object {
            "color": "#595959",
            "fontSize": 0,
          },
          "subPage": Object {
            "color": "#808080",
            "fontSize": 3,
            "lineHeight": "1.6em",
          },
        },
        "variants": Object {
          "avatar": Object {
            "borderRadius": "circle",
            "height": "avatar",
            "width": "avatar",
          },
          "card": Object {
            "bg": "background",
            "boxShadow": "card",
            "p": 2,
          },
          "link": Object {
            "color": "primary",
          },
          "nav": Object {
            ":hover,:focus,.active": Object {
              "color": "primary",
            },
            "color": "inherit",
            "display": "inline-block",
            "fontSize": 1,
            "fontWeight": "bold",
            "p": 2,
            "textDecoration": "none",
          },
        },
      }
    }
  >
    <Text
      variant="default"
    >
      <ForwardRef
        variant="default"
      >
        <styled.div
          tx="text"
          variant="default"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "dKonal",
                  "rules": Array [
                    "box-sizing: border-box;",
                    "margin: 0;",
                    "min-width: 0;",
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "usesTheme": true,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            tx="text"
            variant="default"
          >
            <div
              className="sc-bdVaJa dKonal"
            >
              Default Text
            </div>
          </StyledComponent>
        </styled.div>
      </ForwardRef>
    </Text>
  </ThemeProvider>
</CustomThemeProvider>
`;

exports[`components/Text muted Text 1`] = `
<CustomThemeProvider
  dark={false}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "320px",
          "540px",
          "720px",
          "960px",
          "1140px",
          "1400px",
        ],
        "buttons": Object {
          "outline": Object {
            "bg": "transparent",
            "boxShadow": "inset 0 0 2px",
            "color": "primary",
            "variant": "buttons.primary",
          },
          "primary": Object {
            "bg": "primary",
            "borderRadius": "default",
            "color": "background",
            "fontSize": 2,
            "fontWeight": "bold",
          },
          "secondary": Object {
            "bg": "secondary",
            "color": "background",
            "variant": "buttons.primary",
          },
        },
        "colors": Object {
          "background": "#fff",
          "gray": "#dddddf",
          "highlight": "hsla(205, 100%, 40%, 0.125)",
          "muted": "#f6f6f9",
          "primary": "#07c",
          "secondary": "#30c",
          "text": "#000",
        },
        "fontSizes": Array [
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
        ],
        "fontWeights": Object {
          "body": 400,
          "bold": 900,
          "heading": 700,
        },
        "fonts": Object {
          "body": "Open Sans, Helvetica Neue, Helvetica, sans-serif",
          "heading": "inherit",
          "monospace": "Menlo, monospace",
        },
        "lineHeights": Object {
          "body": 1.5,
          "heading": 1.125,
        },
        "radii": Object {
          "circle": 99999,
          "default": 4,
        },
        "shadows": Object {
          "card": "0 0 4px rgba(0, 0, 0, .125)",
        },
        "sizes": Object {
          "avatar": 48,
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "styles": Object {
          "root": Object {
            "fontFamily": "body",
            "fontWeight": "body",
            "lineHeight": "body",
          },
        },
        "text": Object {
          "bold": Object {
            "color": "#595959",
            "fontWeight": "900",
          },
          "default": Object {
            "color": "#595959",
          },
          "muted": Object {
            "color": "#808080",
          },
          "page": Object {
            "fontSize": 4,
            "letterSpacing": "-0.04em",
            "lineHeight": "1.4em",
            "wordSpacing": "-0.08em",
          },
          "section": Object {
            "fontSize": 3,
            "lineHeight": "1.4em",
          },
          "small": Object {
            "color": "#595959",
            "fontSize": 0,
          },
          "subPage": Object {
            "color": "#808080",
            "fontSize": 3,
            "lineHeight": "1.6em",
          },
        },
        "variants": Object {
          "avatar": Object {
            "borderRadius": "circle",
            "height": "avatar",
            "width": "avatar",
          },
          "card": Object {
            "bg": "background",
            "boxShadow": "card",
            "p": 2,
          },
          "link": Object {
            "color": "primary",
          },
          "nav": Object {
            ":hover,:focus,.active": Object {
              "color": "primary",
            },
            "color": "inherit",
            "display": "inline-block",
            "fontSize": 1,
            "fontWeight": "bold",
            "p": 2,
            "textDecoration": "none",
          },
        },
      }
    }
  >
    <Text
      variant="muted"
    >
      <ForwardRef
        variant="muted"
      >
        <styled.div
          tx="text"
          variant="muted"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "bcJdQb",
                  "rules": Array [
                    "box-sizing: border-box;",
                    "margin: 0;",
                    "min-width: 0;",
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "usesTheme": true,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            tx="text"
            variant="muted"
          >
            <div
              className="sc-bdVaJa bcJdQb"
            >
              Muted Text
            </div>
          </StyledComponent>
        </styled.div>
      </ForwardRef>
    </Text>
  </ThemeProvider>
</CustomThemeProvider>
`;

exports[`components/Text small Text 1`] = `
<CustomThemeProvider
  dark={false}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "320px",
          "540px",
          "720px",
          "960px",
          "1140px",
          "1400px",
        ],
        "buttons": Object {
          "outline": Object {
            "bg": "transparent",
            "boxShadow": "inset 0 0 2px",
            "color": "primary",
            "variant": "buttons.primary",
          },
          "primary": Object {
            "bg": "primary",
            "borderRadius": "default",
            "color": "background",
            "fontSize": 2,
            "fontWeight": "bold",
          },
          "secondary": Object {
            "bg": "secondary",
            "color": "background",
            "variant": "buttons.primary",
          },
        },
        "colors": Object {
          "background": "#fff",
          "gray": "#dddddf",
          "highlight": "hsla(205, 100%, 40%, 0.125)",
          "muted": "#f6f6f9",
          "primary": "#07c",
          "secondary": "#30c",
          "text": "#000",
        },
        "fontSizes": Array [
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
        ],
        "fontWeights": Object {
          "body": 400,
          "bold": 900,
          "heading": 700,
        },
        "fonts": Object {
          "body": "Open Sans, Helvetica Neue, Helvetica, sans-serif",
          "heading": "inherit",
          "monospace": "Menlo, monospace",
        },
        "lineHeights": Object {
          "body": 1.5,
          "heading": 1.125,
        },
        "radii": Object {
          "circle": 99999,
          "default": 4,
        },
        "shadows": Object {
          "card": "0 0 4px rgba(0, 0, 0, .125)",
        },
        "sizes": Object {
          "avatar": 48,
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "styles": Object {
          "root": Object {
            "fontFamily": "body",
            "fontWeight": "body",
            "lineHeight": "body",
          },
        },
        "text": Object {
          "bold": Object {
            "color": "#595959",
            "fontWeight": "900",
          },
          "default": Object {
            "color": "#595959",
          },
          "muted": Object {
            "color": "#808080",
          },
          "page": Object {
            "fontSize": 4,
            "letterSpacing": "-0.04em",
            "lineHeight": "1.4em",
            "wordSpacing": "-0.08em",
          },
          "section": Object {
            "fontSize": 3,
            "lineHeight": "1.4em",
          },
          "small": Object {
            "color": "#595959",
            "fontSize": 0,
          },
          "subPage": Object {
            "color": "#808080",
            "fontSize": 3,
            "lineHeight": "1.6em",
          },
        },
        "variants": Object {
          "avatar": Object {
            "borderRadius": "circle",
            "height": "avatar",
            "width": "avatar",
          },
          "card": Object {
            "bg": "background",
            "boxShadow": "card",
            "p": 2,
          },
          "link": Object {
            "color": "primary",
          },
          "nav": Object {
            ":hover,:focus,.active": Object {
              "color": "primary",
            },
            "color": "inherit",
            "display": "inline-block",
            "fontSize": 1,
            "fontWeight": "bold",
            "p": 2,
            "textDecoration": "none",
          },
        },
      }
    }
  >
    <Text
      variant="small"
    >
      <ForwardRef
        variant="small"
      >
        <styled.div
          tx="text"
          variant="small"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "oNpvL",
                  "rules": Array [
                    "box-sizing: border-box;",
                    "margin: 0;",
                    "min-width: 0;",
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "usesTheme": true,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            tx="text"
            variant="small"
          >
            <div
              className="sc-bdVaJa oNpvL"
            >
              Small Text
            </div>
          </StyledComponent>
        </styled.div>
      </ForwardRef>
    </Text>
  </ThemeProvider>
</CustomThemeProvider>
`;
